keras_cv/layers/preprocessing/preprocessing_utils.py

```diff
@@ -24,3 +25,141 @@ def ensure_tensor(inputs, dtype=None):
     if dtype is not None and inputs.dtype != dtype:
         inputs = tf.cast(inputs, dtype)
     return inputs
+
+
+def check_fill_mode_and_interpolation(fill_mode, interpolation):
+    if fill_mode not in {"reflect", "wrap", "constant", "nearest"}:
+        raise NotImplementedError(
+            "Unknown `fill_mode` {}. Only `reflect`, `wrap`, "
+            "`constant` and `nearest` are supported.".format(fill_mode)
+        )
+    if interpolation not in {"nearest", "bilinear"}:
+        raise NotImplementedError(
+            "Unknown `interpolation` {}. Only `nearest` and "
+            "`bilinear` are supported.".format(interpolation)
+        )
+
+
+def get_rotation_matrix(angles, image_height, image_width, name=None):
+    """Returns projective transform(s) for the given angle(s).
+    Args:
+      angles: A scalar angle to rotate all images by, or (for batches of images)
+        a vector with an angle to rotate each image in the batch. The rank must
+        be statically known (the shape is not `TensorShape(None)`).
+      image_height: Height of the image(s) to be transformed.
+      image_width: Width of the image(s) to be transformed.
+      name: The name of the op.
+    Returns:
+      A tensor of shape (num_images, 8). Projective transforms which can be
+        given to operation `image_projective_transform_v2`. If one row of
+        transforms is [a0, a1, a2, b0, b1, b2, c0, c1], then it maps the
+        *output* point `(x, y)` to a transformed *input* point
+        `(x', y') = ((a0 x + a1 y + a2) / k, (b0 x + b1 y + b2) / k)`,
+        where `k = c0 x + c1 y + 1`.
+    """
+    with backend.name_scope(name or "rotation_matrix"):
+        x_offset = (
+            (image_width - 1)
+            - (tf.cos(angles) * (image_width - 1) - tf.sin(angles) * (image_height - 1))
+        ) / 2.0
+        y_offset = (
+            (image_height - 1)
+            - (tf.sin(angles) * (image_width - 1) + tf.cos(angles) * (image_height - 1))
+        ) / 2.0
+        num_angles = tf.shape(angles)[0]
+        return tf.concat(
+            values=[
+                tf.cos(angles)[:, None],
+                -tf.sin(angles)[:, None],
+                x_offset[:, None],
+                tf.sin(angles)[:, None],
+                tf.cos(angles)[:, None],
+                y_offset[:, None],
+                tf.zeros((num_angles, 2), tf.float32),
+            ],
+            axis=1,
+        )
+
+
+def transform(
```

Comment:
some of these utils already exist here: https://github.com/keras-team/keras-cv/blob/master/keras_cv/utils/preprocessing.py#L213